### DISCUSSION TITLE
Using Nix in Hermes SDK

### DISCUSSION THREAD

**Discussion by soareschen** [informalsystems/hermes-sdk#234] [General]
This discussion is moved from https://github.com/informalsystems/hermes-sdk/issues/214#issuecomment-2019992860.

From @thaodt:

> Im trying to use the command in github workflow file to run tests and I faced this error:
image

> I thought my change was made it failed and I tried to switch into the main branch, delete the flake.lock file, but still failed. Can you please assist further?

It sounds like you might be using an outdated version of Nix. Perhaps you can try upgrading or reinstalling Nix? You can find the upgrade instruction here: https://nixos.org/manual/nix/stable/installation/upgrading.

> After upgrading to the latest nix version, the test command is getting build but then I got the error dial tcp: lookup proxy.golang.org: i/o timeout. The solution is export the env var GOPROXY=direct fixed the issue.

I am not sure if you are using the same executable as provide from the Nix shell. Strictly speaking, the test would work if you have a suitable executable in your `$PATH` environment variable. So you could for example run `make install` from a `gaia` project, and then run the test with that specific version of `gaiad`.

> Changing from `localhost` to `127.0.0.1` makes it worked great.

Could it be that there are issues with your OS network configuration? Does something like `http://localhost:8000` work if you run simple web servers such as `python3 -m http.server`?

**Comment by thaodt** [comment]
Thanks @soareschen for initializing this discussion. I resolved the error before making the PR already.
You were right about nix version upgrade and OS network configuration.
`GOPROXY=direct` didn't resolve the issue with `localhost`. In fact, I did some custom network configuration on another project but I forgot. Changing it back to original made the tests worked fine.
